src/pipx/commands/run.py

```diff
@@ -48,7 +48,7 @@ def maybe_script_content(app: str, is_path: bool) -> Optional[str]:
 
     # Look for a local file first.
     app_path = Path(app)
-    if app_path.exists():
+    if app_path.exists() and app_path.is_file():
```

Comment:
```suggestion
    if app_path.is_file():
```

I believe `is_file` implies `exists`.

https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file

> False is also returned if the path doesn’t exist […].